persistence/src/main/java/cz/fi/muni/pa165/entity/Visit.java

```diff
@@ -80,35 +68,10 @@ public void setNote(String note) {
         this.note = note;
     }
 
-    public Map<Mushroom, Integer> getMushrooms() {
-        return mushroomCount;
-    }
-
-    public void setMushrooms(Map<Mushroom, Integer> map) {
-        mushroomCount = map;
-    }
-
-    public void addMushroom(Mushroom mushroom) {
-        if(!this.mushroomCount.containsKey(mushroom)) {
-            this.mushroomCount.put(mushroom, 1);
-        } else {
-            this.mushroomCount.put(mushroom,
-                    this.mushroomCount.get(mushroom) + 1);
-        }
-
-        mushroom.addVisit(this);
-    }
-
-    public void removeMushroom(Mushroom mushroom) {
-        this.mushroomCount.remove(mushroom);
-        mushroom.removeVisit(this);
-    }
+    public Set<MushroomCount> getMushroomFounds() { return mushroomsFound; }
 
-    public void removeAllMushrooms() {
-        for (Mushroom key : this.mushroomCount.keySet()) {
-            key.removeVisit(this);
-        }
-        this.mushroomCount.clear();
+    public void setMushroomFounds(Set<MushroomCount> mushroomFounds) {
```

Comment:
I think we should not have this method here
or redo the relations that way so we would methods like this :
addMushroomCount(MushroomCount mushroomCount);
removeMushroomCount(MushroomCount mushroomCount);